src/lib/mina_stdlib/minalist.ml

```diff
@@ -0,0 +1,23 @@
+let rec has_length len = function
```

Comment:
I think it would be better to call the module `List`, since the type it operates on is an ordinary OCaml list.

Reply:
You might consider upstreaming these to Jane St `Base`

Reply:
If I name it `List`, then I would tend to add `include Core_kernel.List`. Would you agree to that?

Reply:
> You might consider upstreaming these to Jane St `Base.List`

Will see this. Today's rewrite (it was a draft PR after all ;-) is I think cleaner but relies on a function from the standard library, absent from Base.

Indeed `Base.List` does not contain the whole current `Caml.List`. For example, functions`compare_length_with`, which has simplified the code a lot, is not in `Base.List`. Neither is `compare_lengths`.

Reply:
How about `Lists`? That avoids the conflict with `Core.List`, and avoids the redundant Mina from `Mina_stdlib`.

Reply:
I think `List` is okay - this is already changed in the code.